okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/ConcurrencyLimiters.java

```diff
@@ -204,42 +204,41 @@ public void onDropped() {}
     }
 
     final class DefaultConcurrencyLimiter implements ConcurrencyLimiter {
-
         @GuardedBy("this")
-        private final ThreadWorkQueue<SettableFuture<Limiter.Listener>> waitingRequests = new ThreadWorkQueue<>();
+        private final ThreadWorkQueue<QueuedRequest> waitingRequests = new ThreadWorkQueue<>();
         @GuardedBy("this")
         private SimpleLimiter<Void> limiter;
         @GuardedBy("this")
         private ScheduledFuture<?> timeoutCleanup;
         private final Key limiterKey;
         private final Supplier<SimpleLimiter<Void>> limiterFactory;
-
-        private final SafeArg<Optional<String>> safeArgMethod;
-        private final SafeArg<Optional<String>> safeArgPathTemplate;
+        private final LeakDetector<Limiter.Listener> leakDetector = new LeakDetector<>(Limiter.Listener.class);
 
         DefaultConcurrencyLimiter(Key limiterKey, Supplier<SimpleLimiter<Void>> limiterFactory) {
             this.limiterKey = limiterKey;
             this.limiterFactory = limiterFactory;
             this.limiter = limiterFactory.get();
-            this.safeArgMethod = SafeArg.of("method", limiterKey.method());
-            this.safeArgPathTemplate = SafeArg.of("pathTemplate", limiterKey.pathTemplate());
         }
 
         @Override
         public synchronized ListenableFuture<Limiter.Listener> acquire() {
             SettableFuture<Limiter.Listener> future = SettableFuture.create();
             addSlowAcquireMarker(future);
-            waitingRequests.add(future);
+            waitingRequests.add(new QueuedRequest(future, LeakDetector.maybeCreateStackTrace()));
             processQueue();
             return future;
         }
 
         synchronized void processQueue() {
             while (!waitingRequests.isEmpty()) {
-                log.debug("Limit",
-                        SafeArg.of("limit", limiter.getLimit()),
-                        safeArgMethod,
-                        safeArgPathTemplate);
+                if (log.isDebugEnabled()) {
```

Comment:
is this debug line expensive?

Reply:
not hugely, but felt like... it's a debug line?